analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs

```diff
@@ -21,29 +21,33 @@
 using Microsoft.CodeAnalysis;
 using SonarAnalyzer.Helpers;
 using SonarAnalyzer.SymbolicExecution.Constraints;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
 {
     internal class ConstantCheck : SymbolicCheck
     {
         protected override ProgramState PreProcessSimple(SymbolicContext context) =>
-            ConstraintFromValue(context.Operation.Instance.ConstantValue.Value) is { } constraint
-                ? context.SetOperationConstraint(constraint)
+            context.Operation.Instance.ConstantValue.HasValue
+            && ConstraintFromConstantValue(context.Operation) is { } value
+                ? context.State.SetOperationValue(context.Operation, value)
                 : context.State;
 
         public static SymbolicConstraint ConstraintFromType(ITypeSymbol type) =>
```

Comment:
There is only one call site left for this method:
https://github.com/SonarSource/sonar-dotnet/blob/052952d280582ad3e96d83c28e536ecf3de7230f/analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs#L149-L157

I think that loop should not be there (I think it is related to VB `Dim b as Bool` -> b is initialized to false). Is there a fitting OperationKind for that?
In any case: `ConstraintFromType` does not belong here

Reply:
There's no other way. No initialization operation is present in the CFG. And they don't want to add it: https://github.com/dotnet/roslyn/issues/60056

It's not nice here. And we don't have any nicer place. This one is closest to consts.